internal/kuberesource/mutators.go

```diff
@@ -152,6 +210,30 @@ func AddServiceMesh(
 	return res, retErr
 }
 
+func checkIfDeviceExists(resource any, spec *applycorev1.PodSpecApplyConfiguration, volumeName string) error {
+	// Check for existing volume with unique name.
+	for _, volume := range spec.Volumes {
+		if volume.Name != nil && *volume.Name == volumeName {
+			if volume.EmptyDir != nil {
+				return fmt.Errorf("device %s cannot be of type EmptyDir", *volume.Name)
+			}
+			return nil
```

Comment:
We might be able to exclude more invalid configurations here, because there are not many volume types that can be block devices.